### PR TITLE
Porting of the function pointer restriction feature to variable sensitivity domain branch.

### PR DESCRIPTION
**Pull request by NlightNFotis**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by hannes-steffenhagen-diffblue — CHANGES_REQUESTED** [review]
@thk123 doesn't want this to go into VSD directly

**Review by hannes-steffenhagen-diffblue — APPROVED** [review]

**Comment by hannes-steffenhagen-diffblue** [comment]
Commits from

https://github.com/diffblue/cbmc/pull/5265
https://github.com/diffblue/cbmc/pull/5257
https://github.com/diffblue/cbmc/pull/5262

File: unit/Makefile [hunk 1]
```diff
@@ -23,6 +23,8 @@ SRC += analyses/ai/ai.cpp \
        analyses/variable-sensitivity/last_written_location.cpp \
        big-int/big-int.cpp \
        compound_block_locations.cpp \
+       get_goto_model_from_c_test.cpp \
+       goto-instrument/cover_instrument.cpp \
```
Comment (thk123) [inline unit/Makefile]: Noting this line is different from original commit

Comment (thk123) [reply unit/Makefile]: :no_entry_sign: in fact - this shouldn't be here, the line is included two lines down

**Review by thk123 — COMMENTED** [review]
Partial review: checked aed9c79341be03fd2720064b4575c1c414983da8 "Introduce get_goto_model from a C program for uses in unit tests is ported correctly (#5265)

File: unit/Makefile (hunk 1 above)
Comment (hannes-steffenhagen-diffblue) [reply unit/Makefile]: Good spot, fixed

**Review by thk123 — APPROVED** [review]
Verified all the commits have been ported over identically to how they were merged into develop

**Pull request merged**